File: Chapter 04/C#/Section_4_2_Downward_Dependency/Before/SomeRepository.cs
namespace Chapter04.Section_4_2.Before
{
    // DataAccessLayer.cs 
    public class SomeRepository 
    {
        // This is the violation! A lower layer should not
        // know about an upper layer.
        private PresentationLayer _uiLayer = PresentationLayer.Instance;

        public void UpdateData(int id, string newData)
        {
            Console.WriteLine("(Before Refactoring) Saving data to database...");
            
            // VIOLATION! Calling upwards to the UI Layer
            _uiLayer.UpdateStatusLabel($"(Before Refactoring) Data {id} Saved!");
        }
    }
}

File: Chapter 04/JS/section_4_2_downward_dependency/before/someRepository.js
const PresentationLayer = require('./presentationLayer');

/**
 * ARCHITECTURE WARNING: Upward Dependency Violation.
 */
class SomeRepository {
    constructor() {
        // 🚨 VIOLATION: Lower layer holds reference to upper layer.
        // This makes it impossible to use this repo in a CLI or worker.
        this._uiLayer = PresentationLayer.instance;
    }

    updateData(id, newData) {
        console.log("(Before) Saving data to database...");
        
        // 🚨 VIOLATION: Calling upwards to the UI Layer.
        this._uiLayer.updateStatusLabel(`(Before) Data ${id} Saved!`);
    }
}

module.exports = SomeRepository;